internal/keygen/elgamal.go

```diff
@@ -27,34 +27,39 @@ type ElGamal struct {
 	Y *big.Int // public key
 	G *big.Int // generator
 	P *big.Int // modulus
-	Q *big.Int // order
+	Q *big.Int // (P - 1) / 2
 }
 
 // adapted from https://github.com/dlitz/pycrypto/blob/master/lib/Crypto/PublicKey/ElGamal.py
 func NewElGamal(modulusLength int) (*ElGamal, error) {
 	p, err := emmy.GetSafePrime(modulusLength)
```

Comment:
It would be nice to have 
````go
zero := big.NewInt(0)
one := big.NewInt(1)
two := big.NewInt(2)
````
here and replace all subsequent occurences of `big.NewInt(0)` and `big.NewInt(1)` and `big.NewInt(2)` in the function so that we don't need to instantiate them multiple times in the subsequent `for` loop.